test/dataframe.jl

```diff
@@ -662,4 +662,104 @@ module TestDataFrame
             @test x == y
         end
     end
+
+    @testset "permutecols!" begin
+        column_sets = [
+            (1:5, 2:6, 3:7),
+            (1.0:5.0, 2.0:6.0, 3.0:7.0),
+            ('a':'e', 'b':'f', 'c':'g'),
+            (["foo", "bar", "baz"], ["this", "that", "the other"], ["who", "what", "when"]),
+            ([:who, :what, :when], 'a':'c', ["foo", "bar", "baz"]),
+        ]
+
+        for (a, b, c) in column_sets
```

Comment:
Is it really useful to test all these combinations of columns? The exercized code paths are the same. Also you can directly test the three columns case, the one- and two-column cases aren't that special AFAICT.

Reply:
I have a habit of overdoing it a bit on the tests. I'll scale these back.

Reply:
FWIW I don't particularly mind having a lot of tests even if they're somewhat redundant, unless they contribute significantly to the overall time to run the test suite.

Reply:
I'm not concerned about the runtime, more about the unnecessary complexity/verbosity. I'm all for testing all possible situations where it makes sense, but here there's no reason to test multiple number of columns AFAICT.

Reply:
That's fair.